File: trains-gdx-core/src/main/java/com/github/caay2000/trains/util/RandomGenerator.java
package com.github.caay2000.trains.util;

import java.util.Collection;
import java.util.concurrent.ThreadLocalRandom;

public class RandomGenerator {

    private static final ThreadLocalRandom localRandom = ThreadLocalRandom.current();

    private RandomGenerator() {
    }

    public static int randomPositiveInt(int min, int max) {
        return localRandom.nextInt(min, max + 1);
    }

    public static int randomSignedInt(int min, int max) {
        return localRandom.nextInt(0, Math.abs(max) + Math.abs(min) + 1) - Math.abs(min);
    }

    public static <T> T randomItem(Collection<T> collection) {

        int random = randomPositiveInt(0, collection.size() - 1);
        return collection.stream().skip(random).findFirst().get();
    }
}

File: trains-gdx-core/src/main/kotlin/com/github/caay2000/trains/util/RandomGenerator.kt
package com.github.caay2000.trains.util

import java.util.concurrent.ThreadLocalRandom

object RandomGenerator {
    private val localRandom = ThreadLocalRandom.current()
    fun randomPositiveInt(min: Int, max: Int): Int {
        return localRandom.nextInt(min, max + 1)
    }

    fun randomSignedInt(min: Int, max: Int): Int {
        return localRandom.nextInt(0, Math.abs(max) + Math.abs(min) + 1) - Math.abs(min)
    }

    fun <T> randomItem(collection: Collection<T>): T {
        val random = randomPositiveInt(0, collection.size - 1)


        return return collection.shuffled().take(1)[0];

    }
}